### PR TITLE
Support Spring Data JPA's isPersistable

### LINKED ISSUE #10231: spring-data-jpa: Support org.springframework.data.domain.Persistable.isNew()
**Description**
See https://github.com/spring-projects/spring-data-jpa/blob/2.3.1.RELEASE/src/main/asciidoc/jpa.adoc#entity-state-detection-strategies (point 2)

This is a follow-up to: https://github.com/quarkusio/quarkus/issues/10156#issuecomment-648861189

The Interface is already part of the jars that are shipped with the `spring-data-jpa` Quarkus extension, but the method is not yet evaluated.

PS: Point 3 ( Implementing `EntityInformation`) is probably "missing" as well but the Quarkus extension is _not_ (yet?) shipping the related classes at all so this might be entirely out of scope.

### PR DESCRIPTION
**Pull request by geoand**
Resolves: #10231

### CODE REVIEW

**Comment by famod** [comment]
You resolve tickets quicker than I can comment them with "leave a like if you want this to be implemented". 😄

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/StockMethodsAdder.java [hunk 1]
```diff
@@ -187,6 +199,24 @@ private void generateSave(ClassCreator classCreator, FieldDescriptor entityClass
...
+        if (classInfo.interfaceNames().contains(DotNames.SPRING_DATA_PERSISTABLE)) {
+            return true;
+        }
+
+        DotName superDotName = classInfo.superName();
+        if (superDotName.equals(DotNames.OBJECT)) {
+            return false;
+        }
+
+        return isPersistable(superDotName);
```
Comment (gastaldi) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/StockMethodsAdder.java:217]: Iteration is more efficient than recursion :wink:

Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/StockMethodsAdder.java:217]: No doubt 👍

**Comment by geoand** [comment]
@gsmet can you take a quick look at this one?

**Review by gsmet — APPROVED** [review]

**Pull request merged**